From 10ad9262a184e1afc088bee35b7fa4c188d9d63f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Gr=C3=A4=C3=9Flin?= <mgraesslin@kde.org>
Date: Mon, 19 Oct 2015 17:26:17 +0200
Subject: [PATCH 3/3] [decorations] Delay closeWindow to next event cycle

Once again our famous "closing glxgears crashes Aurorae".

BUG: 346857
FIXED-IN: 5.4.3
REVIEW: 125704
---
 decorations/decoratedclient.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/decorations/decoratedclient.cpp b/decorations/decoratedclient.cpp
index 9ebd241..69b2032 100644
--- a/decorations/decoratedclient.cpp
+++ b/decorations/decoratedclient.cpp
@@ -163,10 +163,14 @@ DELEGATE(requestToggleKeepBelow, KeepBelowOp)
 
 DELEGATE(requestContextHelp, showContextHelp)
 DELEGATE(requestMinimize, minimize)
-DELEGATE(requestClose, closeWindow)
 
 #undef DELEGATE
 
+void DecoratedClientImpl::requestClose()
+{
+    QMetaObject::invokeMethod(m_client, "closeWindow", Qt::QueuedConnection);
+}
+
 QColor DecoratedClientImpl::color(KDecoration2::ColorGroup group, KDecoration2::ColorRole role) const
 {
     auto dp = m_client->decorationPalette();
-- 
2.5.0

